SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorLayoutDetails.cpp

```diff
@@ -133,6 +134,131 @@ void FSpatialGDKEditorLayoutDetails::CustomizeDetails(IDetailLayoutBuilder& Deta
 				SNew(STextBlock).Text(FText::FromString("Push SpatialOS settings to iOS device"))
 			]
 		];
+
+	AssemblyNameInputErrorReporting = SNew(SPopupErrorText);
+	AssemblyNameInputErrorReporting->SetError(TEXT(""));
+	DeploymentNameInputErrorReporting = SNew(SPopupErrorText);
+	DeploymentNameInputErrorReporting->SetError(TEXT(""));
+	IDetailCategoryBuilder& CloudCategory = DetailBuilder.EditCategory("Cloud");
+	CloudCategory.AddCustomRow(FText::FromString("Assembly name"))
+		.NameContent()
+		[
+			SNew(SHorizontalBox)
```

Comment:
I think you could keep the UProperty meta tags and use IDetailPropertyRow::CustomWidget and IDetailPropertyRow::CreatePropertyNameWidget.
This would result in significantly less code while allowing the same amount of customization you need.